### PR TITLE
Document release process in README.md

### PR DESCRIPTION
**Pull request by ChrsMark**
We need to make sure a follow a concrete release process in this repository. In this PR docs are added to list the steps required for a new tag.

@kvch how do you think about Releases? Should we be adding Release along with every tag?

cc: @gizas

### CODE REVIEW

**Comment by ChrsMark** [comment]
I propose we follow sth similar to https://github.com/elastic/elastic-package/blob/main/.ci/Jenkinsfile#L119-L135 + https://github.com/elastic/elastic-package/blob/main/.goreleaser.yml to automate the Release after a tag is created. @gizas any thoughts on this?

**Comment by ChrsMark** [comment]
I see in https://github.com/elastic/elastic-agent-libs/releases releases are made manually too. @cmacknz could you confirm that the team releases manually there? What do you think about unifying and automating the release process?

**Comment by gizas** [comment]
Versioning with goreleaser is based on the fact that our code is go. So what happens in other repos?
So here we are ok and it covers us.

I have used https://github.com/GitTools/GitVersion/releases . So we need to make Jenkins+GitVersion configuration and we keep a numbering and naming based on the git branches

Different logic comparing to goreleaser

File: README.md [hunk 1]
```diff
@@ -7,3 +7,27 @@ This repo contains packages required by autodiscover.
...
+$ git push upstream v0.2.1 
+Enumerating objects: 1, done.
+Counting objects: 100% (1/1), done.
+Writing objects: 100% (1/1), 190 bytes | 190.00 KiB/s, done.
+Total 1 (delta 0), reused 0 (delta 0)
+To https://github.com/elastic/elastic-agent-autodiscover.git
+ * [new tag]             v0.2.1 -> v0.2.1
+```
+
+Then the tag should be available at https://github.com/elastic/elastic-agent-autodiscover/tags and anyone can use the the new version of the library in other projects. For example in order to use `v0.2.1` in Beats projects one would need a `go get github.com/elastic/elastic-agent-autodiscover@v0.2.1`.
```
Comment (cmacknz) [inline README.md:33]: We should mention that the changelog should be updated: https://github.com/elastic/elastic-agent-autodiscover/blob/main/CHANGELOG.md

**Comment by ChrsMark** [comment]
Hmm, I also like the idea of creating the Release manually from GH UI, using a new tag on the fly (it is created automatically with the release). This saves us some automation effort and we make sure that we release on purpose using the proper content from changelog etc.

**Review by cmacknz — APPROVED** [review]

**Pull request merged**